### PR TITLE
Reopen: Support CIDR in rate-limit

### PR DESCRIPTION
**Pull request by spikeekips**
### Background
Followed by #720, this tries to support CIDR.

### CODE REVIEW

**Comment by Charleslee522** [comment]
Oh, `TestCheckInflationBlockIncrease` is failed also here.

**Comment by spikeekips** [comment]
@Charleslee522 Yes, I run it again.

File: lib/network/http_middlewares.go [hunk 1]
```diff
@@ -125,19 +125,23 @@ func RateLimitMiddleware(logger logging.Logger, rule common.RateLimitRule) mux.M
...
 						middleware = middlewares[key]
 						middlewareCache.Add(ip, key)
+						found = true
 						break
 					}
-				}
-			}
 
-			if middleware == nil {
+					if !found {
```
Comment (Charleslee522) [inline lib/network/http_middlewares.go]: @spikeekips `found` is always false here.

Comment (spikeekips) [reply lib/network/http_middlewares.go]: @Charleslee522 Hmm, please see https://github.com/bosnet/sebak/pull/728/files/10dc54737c6ddecc74538f1619a170aeb3634473#diff-516f47ab88d984bfc549e69c298c2d5eR136

Comment (Charleslee522) [reply lib/network/http_middlewares.go]: @spikeekips Oh sorry, just ignore it :)

**Comment by spikeekips** [comment]
Thanks @Charleslee522 👍

**Pull request merged**